packages/utils-query-operation/lib/FragmentSelectorShapes.ts

```diff
@@ -58,7 +64,7 @@ function doesShapeAcceptOperationRecurseShape(
       return shapeOperation.pattern.type === operation.type;
     }
     case 'wildcard': {
-      return true;
+      return operation.type !== Algebra.types.EXPRESSION || operation.expressionType !== Algebra.expressionTypes.NAMED;
```

Comment:
This seems a bit too strict. Any kind of expression will be not allowed through this.